src/pyggdrasil/scripts/cell_simulation.py

```diff
@@ -0,0 +1,369 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Generate cell mutation matrices,
+by generating random trees and sampling cell attachments
+with noise if error rates are non-zero.
+
+As per definition in the SCITE Jahn et al. 2016.
+"""
+
+import argparse
+import jax.random as random
+import numpy as np
+from jax.random import PRNGKeyArray
+import json
+import os
+
+from pyggdrasil.tree import TreeNode
+from pyggdrasil.serialize._to_json import serialize_tree_to_dict
+import pyggdrasil.tree_inference._simulate as sim
+
+
+def create_parser() -> dict:
+    """
+    Parser for required input user.
+
+    Returns:
+        args: dict
+    """
+
+    parser = argparse.ArgumentParser(
+        description="Generate test cell mutation matrices."
+    )
+    parser.add_argument(
+        "--seed",
+        required=False,
+        help="Seed for random generation",
+        type=int,
+        default=42,
+    )
+    parser.add_argument("--outdir", required=True, help="Path to output directory")
+    parser.add_argument("--n_trees", required=True, help="Number of trees", type=int)
+    parser.add_argument("--n_cells", required=True, help="Number of cells", type=int)
+    parser.add_argument(
+        "--n_mutations", required=True, help="Number of mutations", type=int
+    )
+
+    parser.add_argument(
+        "--strategy",
+        required=False,
+        choices=["UNIFORM_INCLUDE_ROOT", "UNIFORM_EXCLUDE_ROOT"],
+        help="Cell attachment strategy",
+        default="UNIFORM_INCLUDE_ROOT",
+    )
+
+    parser.add_argument(
+        "--alpha", required=True, help="False negative rate", type=float
+    )
+    parser.add_argument("--beta", required=True, help="False positive rate", type=float)
+
+    parser.add_argument(
+        "--na_rate", required=True, help="Missing entry rate", type=float
+    )
+
+    parser.add_argument(
+        "--observe_homozygous",
+        required=True,
+        help="Observing homozygous mutations",
+        type=bool,
+    )
+
+    parser.add_argument(
+        "--verbose",
+        required=False,
+        default=False,
+        help="Print trees and full save path",
+        type=bool,
+    )
+
+    args = parser.parse_args()
+    params = vars(args)
+
+    return params
+
+
+def generate_random_tree(rng: PRNGKeyArray, n_nodes: int) -> np.ndarray:
+    """
+    Generates a random tree with n nodes, where the root is the first node.
+    Args:
+        rng: JAX random number generator
+        n_nodes: int number of nodes in the tree
+
+    Returns:
+        adj_matrix: np.ndarray
```

Comment:
I think it'd be good to specify here whether `adj_matrix[i, j]` means an edge "i->j" or the other way around :wink: